graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/aggregations/AggregationsFetcherSupport.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations;
+
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.AVG;
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.COUNT;
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.MAX;
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.MIN;
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.SUM;
+
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.SelectedField;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.query.builder.QueryBuilder;
+import io.stargate.db.schema.Table;
+import io.stargate.graphql.schema.cqlfirst.dml.NameMapping;
+import io.stargate.graphql.schema.cqlfirst.dml.fetchers.DbColumnGetter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+public class AggregationsFetcherSupport {
+
+  private final Table table;
+  private final DbColumnGetter dbColumnGetter;
+
+  public AggregationsFetcherSupport(NameMapping nameMapping, Table table) {
+    this.table = table;
+    this.dbColumnGetter = new DbColumnGetter(nameMapping);
+  }
+
+  public void addAggregationFunctions(
+      DataFetchingEnvironment environment, QueryBuilder.QueryBuilder__20 queryBuilder) {
```

Comment:
Would it be possible to pass the dataStore and return a QueryBuilder? Or Something to that effect? That way we can avoid passing around this autogenerated `QueryBuilder__20` and this will be easier to unit test?

Reply:
I've tried to do it, but the problem is that it needs to operate on the pre-created builder in the `QueryFetcher`:
```

    QueryBuilder.QueryBuilder__20 queryBuilder =
        dataStore.queryBuilder().select().column(buildQueryColumns(environment));
```
I am aware that this is a bit problematic. I will try to find a better way, but I am not sure if we can find a cleaner and better solution.

Reply:
good suggestions. Implemented it: https://github.com/stargate/stargate/pull/901/commits/75551240edd632a75ea51bfb8aea71ae5c23be18